### PR TITLE
[EventStore] Start

### PR DESCRIPTION
**Pull request by tyx**
* [x] Need to lock input and output to a specific class like `NormalizedDomainEvent`

### CODE REVIEW

File: src/Component/EventStore/Infra/Persistence/InMemoryEventStore.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        if (isset($this->aggregates[$streamName])) {
+            if (is_resource($this->aggregates[$streamName])) {
+                fclose($this->aggregates[$streamName]);
+            }
+            unset($this->aggregates[$streamName]);
+        }
+    }
+
+    private function writeEvent(NormalizedDomainEvent $event)
```
Comment (tyx) [inline src/Component/EventStore/Infra/Persistence/InMemoryEventStore.php]: to delete

File: src/Component/EventStore/Domain/EventStreamIterator.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+declare(strict_types=1);
+
+namespace Ubirak\Component\EventStore\Domain;
+
+interface EventStreamIterator extends \Iterator
+{
+    public function current(): ?NormalizedDomainEvent;
```
Comment (shouze) [inline src/Component/EventStore/Domain/EventStreamIterator.php:8]: 👍

File: src/Component/EventStore/Domain/OptimisticConcurrencyConflicted.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
+namespace Ubirak\Component\EventStore\Domain;
+
+class OptimisticConcurrencyConflicted extends \Exception
```
Comment (shouze) [inline src/Component/EventStore/Domain/OptimisticConcurrencyConflicted.php]: extends RuntimeException or LogicException instead?

File: src/Component/EventStore/.atoum.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+This file will automatically be included before EACH run.
+
+Use it to configure atoum or anything that needs to be done before EACH run.
+
+More information on documentation:
+[en] http://docs.atoum.org/en/latest/chapter3.html#configuration-files
+[fr] http://docs.atoum.org/fr/latest/lancement_des_tests.html#fichier-de-configuration
+*/
+
+use mageekguy\atoum;
```
Comment (shouze) [inline src/Component/EventStore/.atoum.php]: this line is useless you can 🔥

File: .travis.yml [hunk 5]
```diff
@@ -81,13 +83,13 @@ install:
       run_tests () {
           set -e
           if [[ $deps = high ]]; then
-              echo "$COMPONENTS" | parallel --gnu -j10% "tfold {} 'cd {} && $COMPOSER_UP && $ATOUM'"
+              echo "$COMPONENTS" | parallel --gnu -j5% "tfold {} 'cd {} && $COMPOSER_UP && $ATOUM'"
```
Comment (tyx) [inline .travis.yml:86]: this is what it fixed travis... need to dig to get why

**Review by shouze — APPROVED** [review]

**Pull request merged**